### DISCUSSION TITLE
Bamboo Mua adapter uses SMTP relay wrong

### DISCUSSION THREAD

**Discussion by thefeli73** [plausible/analytics#4342] [Self-Hosted Support] [answered]
i am unable to set up the SMTP mailer, and so far i have discovered this real issue!

# Observed behavior: 

1. `plausible-conf.env` defines `SMTP_HOST_ADDR=mail.domain.com`
2. plausible tries to send an email
3. bamboo looks up MX of `mail.domain.com`
4. MX of `mail.domain.com` may be something unrelated like `feedback-smtp.eu-north-1.amazonses.com.`, for dmarc monitoring
5. bamboo looks up A record of `feedback-smtp.eu-north-1.amazonses.com.`
6. bamboo tries to establish SMTP connection to non-SMTP server.
7. `{:error, %Mua.TransportError{reason: :timeout}}`

# Expected behavior: 
1. `plausible-conf.env` defines `SMTP_HOST_ADDR=mail.domain.com`
2. plausible tries to send an email
3. bamboo looks up A of `mail.domain.com`
4. bamboo tries to establish SMTP connection to SMTP server which was specified.
5. profit.

**Comment by ruslandoga** [comment, marked as answer]
👋 @thefeli73 

This indeed seems like a wrong way to use SMTP relay. I've created an issue https://github.com/ruslandoga/bamboo_mua/issues/47 and will fix it tomorrow. Thank you for taking time to report it!

**Reply by ruslandoga** [reply]
PR: https://github.com/plausible/analytics/pull/4350

**Comment by ChristopheKar** [comment]
I'm not sure if the introduction of the `Mua` adapter affected other things, but my problem also extends to the `Bamboo.SMTPAdapter` in this new v2.1. I had a previous setup with v2.0 that was working well, but updating to the new version breaks email using both `Bamboo.SMTPAdapter` and `Bamboo.Mua`. 

If I keep the same `docker-compose.yml` file but simply revert the Docker image from `ghcr.io/plausible/community-edition:v2.1` to `plausible/analytics:v2.0`, email works again.

**Reply by ruslandoga** [reply]
👋 @ChristopheKar 

Erlang version changed between v2.0 and v2.1 and the new Erlang version introduced more strict TLS defaults and peer verifications. This broke `Bamboo.SMTPAdapter` since it didn't allow peer verification configuration and Erlang requires it and is the reason for `Bamboo.Mua` introduction. If you share the problems you are having with `Bamboo.Mua` it's very likely we will find a workaround. Based on previous discussions here and in the issues it usually involves some extra TLS configuration and is relatively easy to do.

**Reply by ruslandoga** [reply]
To save time, you can also try searching the discussions for "mua" or "smtp" to find common problems and their workarounds.

**Reply by ChristopheKar** [reply]
My bad, I somehow missed the relevant discussion #4149. One of your suggestions in that conversation worked well for me, thank you for your work and prompt responses!

**Comment by thefeli73** [comment]
i finally fixed my issue. i had fatfingered an ip idress in the firewall for port 587. since i only tried port 465 for the longest time, which still doesnt work, i never thought to check if plausible can reach my mailserver on port 587, since 465 was reachable.
